unity-client/Assets/Scripts/MainScripts/DCL/Controllers/Scene/ParcelScene.cs

```diff
@@ -90,8 +92,9 @@ private void Update()
 
         protected virtual string prettyName => sceneData.basePosition.ToString();
 
-        protected void RefreshName()
+        protected void RefreshSceneState()
```

Comment:
The `RefreshName()` method was intended for debugging purposes, here we are adding another responsibility to it. Maybe its a bit more maintainable to just make `state` a setter and call the event from there?.

```
public int stateValue;
public int state 
{
    get { return stateValue; }
    set 
    {
        if ( stateValue != value )
            OnStateRefreshed?.Invoke(this);  

        stateValue = value;
    } 
}
```

Reply:
You're right, it is better to separate resposibilities. Done! Thanks :)